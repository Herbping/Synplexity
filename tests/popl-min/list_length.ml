type nat =
  | Z
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let rec list_length : list -> nat |>
/\([ ]    -> 0,
   [nat]  -> 1,
   [nat; nat] -> 2) = ?
