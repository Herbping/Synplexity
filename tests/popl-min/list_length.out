(* Generated Thu Oct  8 20:38:31 2015 *)
fix f1 (l1:list) : nat ->
  match l1 with
  | Nil u1 -> Z
  | Cons t1 -> S (f1 (#2 t1))
